python/dgl/data/qm9_edge.py

```diff
@@ -0,0 +1,404 @@
+        If not provided, will load all the labels.
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
+
+    Attributes
+    ----------
+    num_labels : int
+        Number of labels for each graph, i.e. number of prediction tasks
+
+    Raises
+    ------
+    UserWarning
+        If the raw data is changed in the remote server by the author.
+    
+    Examples
+    --------
+    >>> data = QM9EdgeDataset(label_keys=['mu', 'alpha'])
+    >>> data.num_labels
+    2
+    
+    >>> # iterate over the dataset
+    >>> for graph, labels in data:
+    ...     print(graph) # get information of each graph
+    ...     print(labels) # get labels of the corresponding graph
+    ...     # your code here...
+    >>>
+    """
+    
+    raw_url = ('https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/'
+            'molnet_publish/qm9.zip')
+    raw_url2 = 'https://ndownloader.figshare.com/files/3195404'
+    
+    keys = ['mu', 'alpha', 'homo', 'lumo', 'gap', 'r2', 'zpve', 'U0', 'U', 'H', 'G', 'Cv', 'U0_atom',
+            'U_atom', 'H_atom', 'G_atom', 'A', 'B', 'C']
+    map_dict = {}
+    
+    for i, key in enumerate(keys):
+        map_dict[key] = i
+        
+    if rdkit is not None:
+        types = {'H': 0, 'C': 1, 'N': 2, 'O': 3, 'F': 4}
+        bonds = {BT.SINGLE: 0, BT.DOUBLE: 1, BT.TRIPLE: 2, BT.AROMATIC: 3}
+        
+    def __init__(self, 
+                 load_raw=False,
+                 label_keys=None,
+                 raw_dir=None, 
+                 force_reload=False, 
+                 verbose=True):
+        
+        if label_keys == None:
+            self.targets = None
+            self.num_labels = 19
+        else:
+            self.targets = [self.map_dict[i] for i in label_keys]
+            self.num_labels = len(label_keys)
+            
+        self.load_raw = load_raw
+                
+        self._url = _get_dgl_url('dataset/qm9_edge.npz')
+        
+        super(QM9EdgeDataset, self).__init__(name='qm9Edge',
+                                            raw_dir=raw_dir,
+                                            url = self._url,
+                                            force_reload=force_reload,
+                                            verbose=verbose)
+    
+    def download(self):
+        
+        if rdkit is None or not self.load_raw:
+            print('Using a pre-processed version of the dataset.'
+                  'Please install `rdkit` to alternatively process the raw data.')
+            
+            file_path = f'{self.raw_dir}/qm9_edge.npz'
+            if not os.path.exists(file_path):
+                download(self._url, path=file_path)
+    
+        else:
+            if not os.path.exists(os.path.join(self.raw_dir, "gdb9.sdf.csv")):
```

Comment:
It's better to stick to the same style/usage for both branches, e.g. use one of `f'{self.raw_dir}/xxx'` and `os.path.join(self.raw_dir, "xxx")`.